addons/stock/models/stock_move.py

```diff
@@ -709,17 +714,17 @@ def _assign_picking(self):
                     })
             else:
                 recompute = True
-                picking = Picking.create(move._get_new_picking_values())
-
-            move.write({'picking_id': picking.id})
-            move._assign_picking_post_process(new=recompute)
-            # If this method is called in batch by a write on a one2many and
-            # at some point had to create a picking, some next iterations could
-            # try to find back the created picking. As we look for it by searching
-            # on some computed fields, we have to force a recompute, else the
-            # record won't be found.
+                picking = Picking.create(moves._get_new_picking_values())
+
+            moves.write({'picking_id': picking.id})
+            moves._assign_picking_post_process(new=recompute)
             if recompute:
-                move.recompute()
+                # If this method is called in batch by a write on a one2many and
+                # at some point had to create a picking, some next iterations could
+                # try to find back the created picking. As we look for it by searching
+                # on some computed fields, we have to force a recompute, else the
+                # record won't be found.
+                moves.recompute()
```

Comment:
now that you create the picking for a group of move, i'm not sur you still need the recompute logic
please check the test added in the following revison [0] still passes without this logic

[0] https://github.com/odoo/odoo/commit/8fe8ca9948466064675beb3931b457b67ebdedc6